content/en/docs/contribute/generate-ref-docs/kubernetes-components.md

```diff
@@ -118,134 +118,79 @@ need to work with someone who can set the label and milestone for you.
+`src` and you can just provide the directory name as the value for `dst`.
+For example:
 
 ```shell
-- name: federation
-  remote: https://github.com/kubernetes/federation.git
-#  # Change this to a release branch when federation has release branches.
-  branch: master
   files:
-  - src: docs/admin/kubefed_init.md
-    dst: docs/reference/generated/kubefed_init.md
-  ...
+  - src: gen-compdocs/build/kubeadm*.md
+    dst: content/en/docs/reference/setup-tools/kubeadm/generated/
 ```
 
-Here's an example a `config.yaml` file that shows the sources and
-destinations of all the Markdown files that were generated and copied
-by the `update-imported-docs` tool at the beginning of the Kubernetes
-1.9 release.
-
-```shell
-repos:
-- name: kubernetes
-  remote: https://github.com/kubernetes/kubernetes.git
-  branch: release-1.9
-  files:
-  - src: docs/admin/cloud-controller-manager.md
-    dst: docs/reference/generated/cloud-controller-manager.md
-  - src: docs/admin/kube-apiserver.md
-    dst: docs/reference/generated/kube-apiserver.md
-  - src: docs/admin/kube-controller-manager.md
-    dst: docs/reference/generated/kube-controller-manager.md
-  - src: docs/admin/kubelet.md
-    dst: docs/reference/generated/kubelet.md
-  - src: docs/admin/kube-proxy.md
-    dst: docs/reference/generated/kube-proxy.md
-  - src: docs/admin/kube-scheduler.md
-    dst: docs/reference/generated/kube-scheduler.md
-  - src: docs/user-guide/kubectl/kubectl.md
-    dst: docs/reference/generated/kubectl/kubectl.md
-- name: federation
-  remote: https://github.com/kubernetes/federation.git
-#  # Change this to a release branch when federation has release branches.
-  branch: master
-  files:
-  - src: docs/admin/federation-apiserver.md
-    dst: docs/reference/generated/federation-apiserver.md
-  - src: docs/admin/federation-controller-manager.md
-    dst: docs/reference/generated/federation-controller-manager.md
-  - src: docs/admin/kubefed_init.md
-    dst: docs/reference/generated/kubefed_init.md
-  - src: docs/admin/kubefed_join.md
-    dst: docs/reference/generated/kubefed_join.md
-  - src: docs/admin/kubefed.md
-    dst: docs/reference/generated/kubefed.md
-  - src: docs/admin/kubefed_options.md
-    dst: docs/reference/generated/kubefed_options.md
-  - src: docs/admin/kubefed_unjoin.md
-    dst: docs/reference/generated/kubefed_unjoin.md
-  - src: docs/admin/kubefed_version.md
-    dst: docs/reference/generated/kubefed_version.md
-  ```
-
 ## Running the update-imported-docs tool
 
-Now that your `config.yaml` file contains your sources and destinations,
-you can run the `update-imported-docs` tool:
+After having reviewed and/or customized the `reference.yaml` file, you can run
+the `update-imported-docs` tool:
 
 ```shell
-cd <web-base>
-go get ./update-imported-docs
-go run update-imported-docs/update-imported-docs.go
+cd <web-base>/update-imported-docs
+./update-imported-docs reference.yml
```

Comment:
I'm getting the `ImportError: No module named yaml` error as well. And depending on what OS the user is on, there are different ways to install pyyaml. This is a potential friction point.

Reply:
Okay, updating the script to print error message if PyYAML is not installed.

Reply:
Also updated the README.md by making this a requirement.